apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -219,6 +220,27 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 		if r.Spec.Autoscaler != nil && r.Spec.Autoscaler.TargetMemoryUtilization != nil && (*r.Spec.Autoscaler.TargetMemoryUtilization < int32(1) || *r.Spec.Autoscaler.TargetMemoryUtilization > int32(99)) {
 			return fmt.Errorf("the OpenTelemetry Spec autoscale configuration is incorrect, targetMemoryUtilization should be greater than 0 and less than 100")
 		}
+
+		if r.Spec.Autoscaler != nil {
```

Comment:
thought: I'm wondering if we can clean this up a little bit by putting the autoscaler logic in its own method and put it in a single if condition and moving it to the bottom of the function i.e.
```
if r.Spec.Autoscaler != nil {
    return checkAutoscalerSpec(r.Spec.Autoscaler)
}
return nil
```